In [1]:
import pandas as pd
import re

# 0. 설문조사 결과 가져오기

In [2]:
data = pd.read_csv('survey_data.csv')

In [3]:
data.head()

타임스탬프 연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)  \
0  2020/05/18 5:26:50 오후 GMT+9                              20 ~ 24세   
1  2020/05/18 5:43:14 오후 GMT+9                              20 ~ 24세   
2  2020/05/18 5:44:06 오후 GMT+9                              20 ~ 24세   
3  2020/05/18 5:45:14 오후 GMT+9                                19세 이하   
4  2020/05/18 5:46:18 오후 GMT+9                              20 ~ 24세   

  성별을 선택해주세요 기초제품을 구매하거나 선물한 적이 있나요?  \
0         여성                       예   
1         여성                       예   
2         여성                       예   
3         여성                       예   
4         여성                       예   

                평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)  \
0  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
1  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
2  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
3                     자신의 피부에 맞게 상품을 비교하고 선택할 수 있어요.   
4  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   

  기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)  \
0  제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서...      
1     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
2     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
3                                 진짜 좋은 제품인지 알 수 없어서      
4  제품이 너무 다양하고 많아서 정보를 알 수 없어서;피부가 민감하지 않아서 제품 다양...      

           화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)  \
0  제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;온라인 ...   
1                             제품을 직접 테스트하고 크기 등을 보려고   
2          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
3                             제품을 직접 테스트하고 크기 등을 보려고   
4         집이랑 가까워서;그냥 집가다가 생각나면 사는 거고 아니면 쿠팡으로 사는거구~   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
0                               오프라인 매장 방문을 꺼리지 않아요!   
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   
3                               오프라인 매장 방문을 꺼리지 않아요!   
4                               오프라인 매장 방문을 꺼리지 않아요!   

   화장품과 관련하여 유튜브로 정보를 탐색하시나요?  화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)  \
0                         1.0                                0.0   
1                         0.0                                0.0   
2                         3.0                                1.0   
3                         4.0                                4.0   
4                         1.0                                0.0   

                화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?  \
0         언젠가 구매할지도 모르니까;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
1                                            사용하지 않음   
2             유행을 알고 싶어서;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
3  색조, 메이트업베이스 등에 대한 정보를 얻기 위해;기초제품(스킨, 로션, 미스트, ...   
4                             그냥 화장한 얼굴이 예뻐서 대리만족하려구   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)  
0  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
3  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  
4  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...

In [4]:
data.shape

(137, 12)

타임스탬프를 제외한 11개의 질문과 137개의 응답

# 1. 질문 리스트 만들기

In [5]:
ques = list(data.columns)[1:]  # 타임스탬프 제외
ques

['연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)',
 '성별을 선택해주세요',
 '기초제품을 구매하거나 선물한 적이 있나요?',
 '평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)',
 '기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)',
 '화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)',
 '오프라인 매장 방문을 꺼리는 이유는 무엇인가요?',
 '화장품과 관련하여 유튜브로 정보를 탐색하시나요?',
 '화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)',
 '화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?',
 '어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)']

# 2. 응답자 구분하기

relatd = '업계에 종사하고 있어요.'를 선택한 응답자

customers = '업계에 종사하고 있어요'를 선택하지 않은 응답자 (화장품 미구매자 포함)

purchasers = 스킨케어 제품을 구매한 적이 있다고 답한 응답자 (업계 종사자 포함)

In [6]:
related = pd.DataFrame(columns = data.columns)
customers = pd.DataFrame(columns = data.columns)
purchasers = pd.DataFrame(columns = data.columns)

In [7]:
for i, ans in enumerate(data.iloc[:, 4]):
    if type(ans) is not str:
        customers = customers.append(data.iloc[i])
        continue
    purchasers = purchasers.append(data.iloc[i])
    
    if '업계에 종사하고 있어요'in ans:
        related = related.append(data.iloc[i])
    else:
        customers = customers.append(data.iloc[i])

In [8]:
related.shape

(3, 12)

In [9]:
customers.shape

(134, 12)

In [10]:
purchasers.shape

(124, 12)

# 3. 오프라인 매장 방문을 꺼리는 이유 살피기

답변별로 몇 명이 응답하지는지에 대한 함수 만들기

데이터 프레임과 질문의 번호를 전달하면 '답변 내용' : '인원수'를 반환

In [11]:
for i, q in enumerate(ques):
    print(f'{i+1} : {q}')

1 : 연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)
2 : 성별을 선택해주세요
3 : 기초제품을 구매하거나 선물한 적이 있나요?
4 : 평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)
5 : 기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)
6 : 화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)
7 : 오프라인 매장 방문을 꺼리는 이유는 무엇인가요?
8 : 화장품과 관련하여 유튜브로 정보를 탐색하시나요?
9 : 화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)
10 : 화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?
11 : 어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)


In [12]:
# 더 간단하게 할 수 있는 방법 모색하기

def count_answers(df, ques_num):
    count = dict()
    
    for answer in df.iloc[:,ques_num]:
        if ';' in answer:
            for ans in answer.split(';'):
                if ans not in count.keys():
                    count[ans] = 1
                else:
                    count[ans] += 1
        else:
            if answer not in count.keys():
                count[answer] = 1
            else:
                count[answer] += 1
    return count

In [13]:
count_answers(purchasers, 7)

{'오프라인 매장 방문을 꺼리지 않아요!': 59,
 '온라인 쇼핑에 익숙해져서': 20,
 '혼자 비교하고 결정하고 싶어서': 20,
 '직원들의 접객이 불편해서': 28,
 '혜택이 더 다양하고 많아서': 16,
 '집에서 나가고 싶지 않아서': 16,
 '온라인이 더 싸요': 1,
 '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서': 18,
 '사람 많은 곳을 좋아하지 않아서': 2,
 '귀찮아서': 1,
 '온라인으로 더 저렴히 구매가능해서': 1,
 '온라인으로 받을 수 있는 혜택(ex.할인)이 많아서': 1,
 '나가기귀찮아서': 1}

오프라인 방문을 꺼리지 않는다고 답한 사람들이 꺼리는 이유를 선택한 경우도 있었다.

선택의 이유는 아래의 경우라고 추정할 수 있다.

1. 본인은 오프라인 방문을 꺼리지 않지만, 사람들이 생각한 이유에 대해서 추측한 답변
2. 오프라인 매장 방문을 꺼리지는 않지만, 한 번씩 불편함을 느끼는 정도
3. 업계 종사자가 느낀 일반 고객들의 생각
4. 기타

중복된 답변은 아래와 같이 처리하자.

1. '귀찮'이라는 답변이 들어간 내용은 '집에서 나가고 싶지 않아서'로 합친다.
2. '온라인', '혜택', '저렴'이라는 단어가 들어간 답변은 '온라인에서의 혜택이 더 다양하고 많아서'라는 답변으로 고친다.
3. 기타(추가 사항이 있다면 기록해두기)

In [14]:
temp_counting = count_answers(purchasers, 7)

In [15]:
for key, value in temp_counting.items():
    if '귀찮' in key:
        temp_counting['집에서 나가고 싶지 않아서'] += value
        temp_counting[key] = 0
    elif re.search('온라인.+(혜택|싸[요다서]|저렴)', key):
        temp_counting['혜택이 더 다양하고 많아서'] += value
        temp_counting[key] = 0

temp_counting

{'오프라인 매장 방문을 꺼리지 않아요!': 59,
 '온라인 쇼핑에 익숙해져서': 20,
 '혼자 비교하고 결정하고 싶어서': 20,
 '직원들의 접객이 불편해서': 28,
 '혜택이 더 다양하고 많아서': 19,
 '집에서 나가고 싶지 않아서': 18,
 '온라인이 더 싸요': 0,
 '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서': 18,
 '사람 많은 곳을 좋아하지 않아서': 2,
 '귀찮아서': 0,
 '온라인으로 더 저렴히 구매가능해서': 0,
 '온라인으로 받을 수 있는 혜택(ex.할인)이 많아서': 0,
 '나가기귀찮아서': 0}

In [16]:
temp_counting['온라인 구매가 혜택이 더 다양하고 많아서'] = temp_counting.pop('혜택이 더 다양하고 많아서')
temp_counting

{'오프라인 매장 방문을 꺼리지 않아요!': 59,
 '온라인 쇼핑에 익숙해져서': 20,
 '혼자 비교하고 결정하고 싶어서': 20,
 '직원들의 접객이 불편해서': 28,
 '집에서 나가고 싶지 않아서': 18,
 '온라인이 더 싸요': 0,
 '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서': 18,
 '사람 많은 곳을 좋아하지 않아서': 2,
 '귀찮아서': 0,
 '온라인으로 더 저렴히 구매가능해서': 0,
 '온라인으로 받을 수 있는 혜택(ex.할인)이 많아서': 0,
 '나가기귀찮아서': 0,
 '온라인 구매가 혜택이 더 다양하고 많아서': 19}

In [17]:
reasons_mind_offline = dict()

for k, v in temp_counting.items():
    if v:
        reasons_mind_offline[k] = v

reasons_mind_offline

{'오프라인 매장 방문을 꺼리지 않아요!': 59,
 '온라인 쇼핑에 익숙해져서': 20,
 '혼자 비교하고 결정하고 싶어서': 20,
 '직원들의 접객이 불편해서': 28,
 '집에서 나가고 싶지 않아서': 18,
 '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서': 18,
 '사람 많은 곳을 좋아하지 않아서': 2,
 '온라인 구매가 혜택이 더 다양하고 많아서': 19}

In [18]:
'''
오프라인 매장 방문을 조금이라도 꺼리는 사람들의 비율
     = "오프라인 방문을 꺼리지 않아요"를 선택하지 않은 사람들
        + 꺼리지 않지만, 이유는 알겠는 사람들
구매 경험이 있는 사람들(purchaser) 중에서 살피기
'''

num_mind_visiting, num_prefer_online, num_dislike_hospitality = 0, 0, 0

for ans in purchasers.iloc[:, 7]:
    if re.search('오프라인 매장 방문을 꺼리지 않아요!', ans):
        if ';' not in ans:  # 오프라인 매장을 꺼리는 이유를 전혀 선택하지 않은 사람들
            continue
    num_mind_visiting += 1
    if re.search('온라인', ans):
        num_prefer_online += 1
    elif re.search('(혼자|직원(접객)?)', ans):
        num_dislike_hospitality += 1
        
print(f'오프라인 매장 방문을 조금이라도 꺼리는 고객의 비율 : {round(num_mind_visiting*100/purchasers.shape[0], 2)}%')

# 오프라인을 조금이라도 꺼리는 사람들 중 온라인을 선호기 때문에 굳이 오프라인에 방문하지 않는 비율 찾기
print(f'온라인을 선호하기 떄문에 굳이 매장에 오지 않는 고객의 비율 : {round(num_prefer_online*100/num_mind_visiting, 2)}%')

print(f'접객이 불편하거나 혼자 의사결정을 하고 싶은 고객의 비율 : {round(num_dislike_hospitality*100/num_mind_visiting, 2)}%')

오프라인 매장 방문을 조금이라도 꺼리는 고객의 비율 : 60.48%
온라인을 선호하기 떄문에 굳이 매장에 오지 않는 고객의 비율 : 30.67%
접객이 불편하거나 혼자 의사결정을 하고 싶은 고객의 비율 : 44.0%


접객이 불편하거나 혼자 의사결정을 하고 싶은 고객의 정보 좀 더 살피기

1. 연령대
2. 성별
3. 후보군 추천 희망 비율

In [19]:
print(num_dislike_hospitality)

33


7번째 질문에서 혼자 의사결정 혹은 접객 불편을 선택했다면

그사람의 연령대, 성별, 어떻게 하면~의 응답(11)을 기록

In [20]:
df_purchasers = purchasers[[ques[0], ques[1], ques[6], ques[10]]]
df_purchasers.head()

연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값) 성별을 선택해주세요  \
0                              20 ~ 24세         여성   
1                              20 ~ 24세         여성   
2                              20 ~ 24세         여성   
3                                19세 이하         여성   
4                              20 ~ 24세         여성   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
0                               오프라인 매장 방문을 꺼리지 않아요!   
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   
3                               오프라인 매장 방문을 꺼리지 않아요!   
4                               오프라인 매장 방문을 꺼리지 않아요!   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)  
0  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
3  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  
4  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...

In [21]:
ages = {}
gender = {'여성' : 0, '남성' : 0, '기타' : 0, '응답을 원하지 않음' : 0}
like_recomendation_sys = 0
basic_ways = [
    '직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품을 사는 것이 마음 편해요 등)',
'비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 싶어요 등)',
'특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 : A,B,C / 미백라인 : x,y,z 등)',
'응답을 원하지 않아요']
other_ways = []

for age in list(df_purchasers.iloc[:, 0].unique()):
    ages[age] = 0
    
i = 0
for reason in df_purchasers.iloc[:, 2]:
    if re.search('(혼자|직원(접객)?)', reason):
        ages[df_purchasers.iloc[i, 0]] += 1
        gender[df_purchasers.iloc[i, 1]] += 1
        
        if re.search('(요약|정리)', df_purchasers.iloc[i, 3]):
            like_recomendation_sys += 1
    i += 1
    
    
basic_ways = [
    '직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품을 사는 것이 마음 편해요 등)',
'비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 싶어요 등)',
'특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 : A,B,C / 미백라인 : x,y,z 등)',
'응답을 원하지 않아요']
other_ways = []
for way in df_purchasers.iloc[:, 3]:
    if ';' in way:
        for w in way.split(';'):
            if w not in basic_ways:
                other_ways.append(w)
    else:
        if way not in basic_ways:
            other_ways.append(way)

In [26]:
print ('* 접객이 불편하다고 답한 사람들의 나이 분포')
print(ages)
print('\n')

print('* 접객이 불편하다고 답한 사람들의 성별 분포')
print(gender)
print('\n')

print('* 기초 제품 구매에 있어 원하는 방식에 대한 의견')
print(other_ways)
print('\n')

like_recomendation_sys_whole = 0

for ans in df_purchasers.iloc[:, 3]:
    if re.search('(요약|정리)', ans):
        like_recomendation_sys_whole += 1

print(f'전체 응답자 중에서 요약이나 정리를 선호하는 사람들은 {like_recomendation_sys_whole}명으로, 약 {round(like_recomendation_sys_whole*100/df_purchasers.shape[0], 2)}%이다.')

* 접객이 불편하다고 답한 사람들의 나이 분포
{'20 ~ 24세': 21, '19세 이하': 6, '25 ~ 29세': 9, '40 ~ 44세': 2, '30 ~ 34세': 2, '35 ~ 39세': 0, '응답을 원하지 않음': 0, '45 ~ 49세': 0, '50 ~ 59세': 1}


* 접객이 불편하다고 답한 사람들의 성별 분포
{'여성': 34, '남성': 7, '기타': 0, '응답을 원하지 않음': 0}


* 기초 제품 구매에 있어 원하는 방식에 대한 의견
['사실 특징별로 상품요약은 잘되어있는편이라고 생각하는데 특히 화장품업계에서 무분별한 광고가 오히려 소비자들에게 혼란을 준다고 생각합니당', "보통 하나의 브랜드별로는 분류 및 표기가 잘 되어있는데 사실 매장가서 고민하고 검색하는건 여러 브랜드간의 여드름진정라인 비교를 하는것 같아요. 직원분께 추천받는것도 한 브랜드내에서 제품군을 비교받기보단 모든 브랜드 통틀어서 내가 원하는 성분을 포함한 것 중 가성비가 좋거나 '효과'좋은 걸 추천받고 싶으니까요!", '특징별 요약 + 상품별 비교가 잘 되어있으면 편할 거 같습니다', '알레르기 유발성분을 정리 해주세요']


전체 응답자 중에서 요약이나 정리를 선호하는 사람들은 104명으로, 약 83.87%이다.


In [30]:
age_20s = 0; age_40_over = 0
dislike_contact_20s = 0; dislike_contact_40s = 0
prefer_untact_20s = 0; prefer_untact_40s = 0

i = 0
for age in df_purchasers.iloc[:, 0]:
    if re.search('^2', age):
        age_20s += 1
        if re.search('(접객|혼자)', df_purchasers.iloc[i, 2]):
            dislike_contact_20s += 1
        if re.search('(비대면|요약|정리)', df_purchasers.iloc[i, 3]):
            prefer_untact_20s += 1
    elif re.search('^[456]', age):
        age_40_over += 1
        if re.search('(접객|혼자)', df_purchasers.iloc[i, 2]):
            dislike_contact_40s += 1
        if re.search('(비대면|요약|정리)', df_purchasers.iloc[i, 3]):
            prefer_untact_40s += 1
    i += 1

print(f'접객이 불편하거나 혼자 보고 싶은 20대는 {dislike_contact_20s}명으로 20대 중 약 {round(dislike_contact_20s*100/age_20s, 2)}%를 차지한다.')
print(f'또한, 비대면 응대를 선호하거나 상품의 요약 정리를 원하는 20대는 {prefer_untact_20s}명으로 20대 중 약 {round(prefer_untact_20s*100/age_20s, 2)}%를 차지한다.')
print(f'접객이 불편하거나 혼자 보고 싶은 40대는 {dislike_contact_40s}명으로 40대 중 약 {round(dislike_contact_40s*100/age_40_over, 2)}%를 차지한다.')
print(f'또한, 비대면 응대를 선호하거나 상품의 요약 정리를 원하는 40대는 {prefer_untact_40s}명으로 40대 중 약 {round(prefer_untact_40s*100/age_40_over, 2)}%를 차지한다.')

접객이 불편하거나 혼자 보고 싶은 20대는 30명으로 20대 중 약 37.04%를 차지한다.
또한, 비대면 응대를 선호하거나 상품의 요약 정리를 원하는 20대는 75명으로 20대 중 약 92.59%를 차지한다.
접객이 불편하거나 혼자 보고 싶은 40대는 3명으로 40대 중 약 18.75%를 차지한다.
또한, 비대면 응대를 선호하거나 상품의 요약 정리를 원하는 40대는 12명으로 40대 중 약 75.0%를 차지한다.


# 4. 오프라인 매장 방문 목적 살피기

In [32]:
reason_visit = {'급해서 혹은 빨리 사용하고 싶어서' : 0,
               '온라인은 신뢰할 수 없어서' : 0}

for reason in purchasers.iloc[:,6]:
    if '방문하지 않아요' == reason:
        continue
    elif ';' in reason:
        for r in reason.split(';'):
            if r == '방문하지 않아요':
                continue
            elif re.search('(급해서|당장|빨리)', r):
#                 print(r)
                reason_visit['급해서 혹은 빨리 사용하고 싶어서'] += 1
                reason_visit[r] = 0
                continue
            elif re.search('온라인.*(신뢰|믿을\s?수)', r):
                reason_visit['온라인은 신뢰할 수 없어서'] += 1
                reason_visit[r] = 0
                continue
            if r not in reason_visit.keys():
                reason_visit[r] = 1
            else:
                reason_visit[r] += 1

reason_visit_store = dict()

for k, v in reason_visit.items():
    if re.search('(그냥|생각이?/s나면)', k):
        reason_visit_store['특별한 이유 없음'] = v
        continue
    elif v:
        reason_visit_store[k] = v
        
reason_visit_store

{'급해서 혹은 빨리 사용하고 싶어서': 3,
 '온라인은 신뢰할 수 없어서': 2,
 '제품을 직접 테스트하고 크기 등을 보려고': 44,
 '사람에게 물어보고 구매하고 싶어서': 31,
 '온라인 쇼핑이 익숙하지 않아서': 8,
 '온라인 결제가 귀찮아서/불편해서': 15,
 '집이랑 가까워서': 24,
 '특별한 이유 없음': 1,
 '샘플, 증정품 등 혜택이 더 다양하고 많아서': 10}

# 5. 일반 구매자들의 성별 분포 살피기

In [33]:
purchaser_genders = dict()
purchaser_genders['여성'] = 0
purchaser_genders['남성'] = 0

for gender in purchasers.iloc[:,2]:
    if gender == '여성':
        purchaser_genders[gender] += 1
    elif gender == '남성':
        purchaser_genders[gender] += 1
    else:
        continue

purchaser_genders

{'여성': 91, '남성': 31}

# 6. 스킨케어 지식이 부족한 사람들의 응답 살피기

In [46]:
purchasers['기초 제품 구매가 가능한가?'] = purchasers[ques[3]].apply(lambda x: 'choosable' if re.search('(업계|할 수 있어요)', x) else 'unchoosable')
purchasers.head()

타임스탬프 연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)  \
0  2020/05/18 5:26:50 오후 GMT+9                              20 ~ 24세   
1  2020/05/18 5:43:14 오후 GMT+9                              20 ~ 24세   
2  2020/05/18 5:44:06 오후 GMT+9                              20 ~ 24세   
3  2020/05/18 5:45:14 오후 GMT+9                                19세 이하   
4  2020/05/18 5:46:18 오후 GMT+9                              20 ~ 24세   

  성별을 선택해주세요 기초제품을 구매하거나 선물한 적이 있나요?  \
0         여성                       예   
1         여성                       예   
2         여성                       예   
3         여성                       예   
4         여성                       예   

                평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)  \
0  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
1  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
2  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
3                     자신의 피부에 맞게 상품을 비교하고 선택할 수 있어요.   
4  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   

  기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)  \
0  제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서...      
1     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
2     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
3                                 진짜 좋은 제품인지 알 수 없어서      
4  제품이 너무 다양하고 많아서 정보를 알 수 없어서;피부가 민감하지 않아서 제품 다양...      

           화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)  \
0  제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;온라인 ...   
1                             제품을 직접 테스트하고 크기 등을 보려고   
2          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
3                             제품을 직접 테스트하고 크기 등을 보려고   
4         집이랑 가까워서;그냥 집가다가 생각나면 사는 거고 아니면 쿠팡으로 사는거구~   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
0                               오프라인 매장 방문을 꺼리지 않아요!   
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   
3                               오프라인 매장 방문을 꺼리지 않아요!   
4                               오프라인 매장 방문을 꺼리지 않아요!   

   화장품과 관련하여 유튜브로 정보를 탐색하시나요?  화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)  \
0                         1.0                                0.0   
1                         0.0                                0.0   
2                         3.0                                1.0   
3                         4.0                                4.0   
4                         1.0                                0.0   

                화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?  \
0         언젠가 구매할지도 모르니까;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
1                                            사용하지 않음   
2             유행을 알고 싶어서;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
3  색조, 메이트업베이스 등에 대한 정보를 얻기 위해;기초제품(스킨, 로션, 미스트, ...   
4                             그냥 화장한 얼굴이 예뻐서 대리만족하려구   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능) 기초 제품 구매가 가능한가?  
0  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...       choosable  
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...     unchoosable  
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...     unchoosable  
3  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...       choosable  
4  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...     unchoosable

In [48]:
the_unchoosable = purchasers[purchasers['기초 제품 구매가 가능한가?'] == 'unchoosable']
the_unchoosable

타임스탬프 연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)  \
1     2020/05/18 5:43:14 오후 GMT+9                              20 ~ 24세   
2     2020/05/18 5:44:06 오후 GMT+9                              20 ~ 24세   
4     2020/05/18 5:46:18 오후 GMT+9                              20 ~ 24세   
5     2020/05/18 5:47:56 오후 GMT+9                              20 ~ 24세   
7     2020/05/18 5:55:37 오후 GMT+9                              25 ~ 29세   
..                            ...                                   ...   
125  2020/05/19 12:34:52 오후 GMT+9                              50 ~ 59세   
127   2020/05/19 1:12:52 오후 GMT+9                              25 ~ 29세   
128   2020/05/19 1:31:07 오후 GMT+9                              25 ~ 29세   
132   2020/05/19 6:21:44 오후 GMT+9                              40 ~ 44세   
135  2020/05/20 11:25:19 오후 GMT+9                              20 ~ 24세   

    성별을 선택해주세요 기초제품을 구매하거나 선물한 적이 있나요?  \
1           여성                       예   
2           여성                       예   
4           여성                       예   
5           남성                       예   
7           여성                       예   
..         ...                     ...   
125         여성                       예   
127         남성                       예   
128         남성                       예   
132         여성                       예   
135         여성                       예   

                  평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)  \
1    비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
2    건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
4    비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
5                           건성, 복합성, 지성이 무엇인지는 알고 있어요.   
7                           건성, 복합성, 지성이 무엇인지는 알고 있어요.   
..                                                 ...   
125  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
127                         건성, 복합성, 지성이 무엇인지는 알고 있어요.   
128                         건성, 복합성, 지성이 무엇인지는 알고 있어요.   
132                              나의 피부가 건성인지 지성인지 몰라요.   
135  나의 피부가 건성인지 지성인지 몰라요.;비타민이 미백에 좋다는 것, 시카와 티트리 ...   

    기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)  \
1       제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
2       제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
4    제품이 너무 다양하고 많아서 정보를 알 수 없어서;피부가 민감하지 않아서 제품 다양...      
5                                   진짜 좋은 제품인지 알 수 없어서      
7    제품이 너무 다양하고 많아서 정보를 알 수 없어서;매번 실패해서;진짜 좋은 제품인지...      
..                                                 ...      
125                                 진짜 좋은 제품인지 알 수 없어서      
127                               피부(스킨케어)에 대한 지식이 없어서      
128                                    없음;물어보면 잘 알려주던뎅      
132     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
135                                 진짜 좋은 제품인지 알 수 없어서      

             화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)  \
1                               제품을 직접 테스트하고 크기 등을 보려고   
2            제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
4           집이랑 가까워서;그냥 집가다가 생각나면 사는 거고 아니면 쿠팡으로 사는거구~   
5            제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
7    제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;집이랑 ...   
..                                                 ...   
125  사람에게 물어보고 구매하고 싶어서;온라인 쇼핑이 익숙하지 않아서;온라인 결제가 귀찮...   
127                                 사람에게 물어보고 구매하고 싶어서   
128  제품을 직접 테스트하고 크기 등을 보려고;온라인 결제가 귀찮아서/불편해서;샘플, 증...   
132                                 사람에게 물어보고 구매하고 싶어서   
135          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   

                            오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
1    오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                         직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   
4                                 오프라인 매장 방문을 꺼리지 않아요!   
5                                       집에서 나가고 싶지 않아서   
7                                 오프라인 매장 방문을 꺼리지 않아요!   
..                                                 ...   
125                               오프라인 매장 방문을 꺼리지 않아요!   
127                            

In [49]:
# 스킨 케어 제품을 혼자 결정할 수 없는 사람의 비율
print(f'스킨 케어 제품을 혼자 결정할 수 없는 사람은 전체 응답자 {data.shape[0]}명 중 {the_unchoosable.shape[0]}명이고, 전체의 약{round(the_unchoosable.shape[0]*100/data.shape[0], 2)}%이다.')

스킨 케어 제품을 혼자 결정할 수 없는 사람은 전체 응답자 137명 중 72명이고, 전체의 약52.55%이다.


In [50]:
def classify_reason(res, regex, boolean='False'):
    if re.search(regex, res):
        boolean = 'True'
    return boolean

In [53]:
# 혼자 결정할 수 없는 사람들 중 기초제품 구매에 어려움을 겪은 이유가 지식 부족인 비율 알아보기
the_unchoosable['unchoosable reason'] = the_unchoosable[ques[4]].apply(lambda x: 'knowledge' if classify_reason(x, '(지식.*(없[어다]|부족)|몰라)') else 'others')
print(f'{round(the_unchoosable[the_unchoosable["unchoosable reason"] == "knowledge"].shape[0]*100/the_unchoosable.shape[0], 2)}%가 스킨 케어 지식 부족으로 구매에 어려움을 겪은 적이 있다고 말했다.')

100.0%가 스킨 케어 지식 부족으로 구매에 어려움을 겪은 적이 있다고 말했다.


C:\Users\sorye\anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [54]:
count_answers(the_unchoosable, 7)

{'오프라인 매장 방문을 꺼리지 않아요!': 32,
 '온라인 쇼핑에 익숙해져서': 10,
 '혼자 비교하고 결정하고 싶어서': 8,
 '직원들의 접객이 불편해서': 18,
 '혜택이 더 다양하고 많아서': 6,
 '집에서 나가고 싶지 않아서': 12,
 '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서': 8,
 '귀찮아서': 1,
 '온라인으로 더 저렴히 구매가능해서': 1}

In [55]:
count_answers(the_unchoosable, 11)

{'비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 싶어요 등)': 14,
 '직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품을 사는 것이 마음 편해요 등)': 16,
 '특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 : A,B,C / 미백라인 : x,y,z 등)': 60,
 "보통 하나의 브랜드별로는 분류 및 표기가 잘 되어있는데 사실 매장가서 고민하고 검색하는건 여러 브랜드간의 여드름진정라인 비교를 하는것 같아요. 직원분께 추천받는것도 한 브랜드내에서 제품군을 비교받기보단 모든 브랜드 통틀어서 내가 원하는 성분을 포함한 것 중 가성비가 좋거나 '효과'좋은 걸 추천받고 싶으니까요!": 1,
 '응답을 원하지 않아요': 2,
 '알레르기 유발성분을 정리 해주세요': 1}